gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -330,70 +344,76 @@ private static void setupConjureTypescriptProject(
                             task.workingDir(srcDirectory);
                             task.dependsOn(compileConjureTypeScript);
                             task.getInputs().file(new File(srcDirectory, "package.json"));
                             task.getOutputs().dir(new File(srcDirectory, "node_modules"));
                         });
-                Task compileTypeScript = project.getTasks().create("compileTypeScript", Exec.class, task -> {
-                    task.setDescription("Runs `npm tsc` to compile generated TypeScript files into JavaScript files.");
-                    task.setGroup(TASK_GROUP);
-                    task.commandLine(npmCommand, "run-script", "build");
-                    task.workingDir(srcDirectory);
-                    task.dependsOn(installTypeScriptDependencies);
-                    task.getOutputs().dir(srcDirectory);
-                });
-                Task publishTypeScript = project.getTasks().create("publishTypeScript", Exec.class, task -> {
-                    task.setDescription("Runs `npm publish` to publish a TypeScript package "
-                            + "generated from your Conjure definitions.");
-                    task.setGroup(TASK_GROUP);
-                    task.commandLine(npmCommand, "publish");
-                    task.workingDir(srcDirectory);
-                    task.dependsOn(compileConjureTypeScript);
-                    task.dependsOn(compileTypeScript);
+                TaskProvider<Exec> compileTypeScript = project.getTasks()
+                        .register("compileTypeScript", Exec.class, task -> {
+                            task.setDescription(
+                                    "Runs `npm tsc` to compile generated TypeScript files into JavaScript files.");
+                            task.setGroup(TASK_GROUP);
+                            task.commandLine(npmCommand, "run-script", "build");
+                            task.workingDir(srcDirectory);
+                            task.dependsOn(installTypeScriptDependencies);
+                            task.getOutputs().dir(srcDirectory);
+                        });
+                TaskProvider<Exec> publishTypeScript = project.getTasks()
+                        .register("publishTypeScript", Exec.class, task -> {
+                            task.setDescription("Runs `npm publish` to publish a TypeScript package "
+                                    + "generated from your Conjure definitions.");
+                            task.setGroup(TASK_GROUP);
+                            task.commandLine(npmCommand, "publish");
+                            task.workingDir(srcDirectory);
+                            task.dependsOn(compileConjureTypeScript);
+                            task.dependsOn(compileTypeScript);
+                        });
+                subproj.getPluginManager().withPlugin("maven-publish", _packagingPlugin -> {
```

Comment:
This plugin guard-clause is new, right? Makes sense though.

Reply:
I think that this is a bit of a behaviour change. I'm not sure if we actually apply the `maven-publish` plugin on conjure projects

Reply:
that was a fear of mine. Are there other known sources of the publish task? Or do I have to restructure to check if the task is present? I do that in another place with ideaModule, so can do it here as well if necessary.